op-service/client/http.go

```diff
@@ -0,0 +1,53 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	DefaultTimeoutSeconds = 30
+)
+
+type HTTP interface {
+	Get(ctx context.Context, path string, headers http.Header) (*http.Response, error)
+}
+
```

Comment:
We should add a compile time assertion that the `BasicHTTPClient` implements the `HTTP` interface

Reply:
added.